In [1]:
import pandas as pd
import geopandas as gpd
import warnings

/home/kevin/.local/lib/python3.8/site-packages/geopandas/_compat.py:124: UserWarning: The Shapely GEOS version (3.11.1-CAPI-1.17.1) is incompatible with the GEOS version PyGEOS was compiled with (3.10.4-CAPI-1.16.2). Conversions between both will be slow.
  warnings.warn(
/tmp/ipykernel_1577/1740704329.py:2: DeprecationWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas still uses PyGEOS by default. However, starting with version 0.14, the default will switch to Shapely. To force to use Shapely 2.0 now, you can either uninstall PyGEOS or set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:

import os
os.environ['USE_PYGEOS'] = '0'
import geopandas

In the next release, GeoPandas will switch to using Shapely by default, even if PyGEOS is installed. If you only have PyGEOS installed to get speed-ups, this switch should be smooth. However, if you are using PyGEOS directly (

In [2]:
warnings.filterwarnings('ignore') # hide warnings

https://discover.data.vic.gov.au/dataset/vicmap-admin-local-government-area-lga-polygon-aligned-to-property

In [3]:
# Read the shapefile
shapefile_path = 'data/raw/VMADMIN/LGA_POLYGON.shp'
gdf = gpd.read_file(shapefile_path)

# Display the first few records to understand its structure
gdf.head()


UFI  PFI LGA_CODE       LGA_NAME             GAZ_LGA  \
0  856969541  308      308       BRIMBANK       BRIMBANK CITY   
1  820277766  356      356      NILLUMBIK     NILLUMBIK SHIRE   
2  820277774  311      311       CARDINIA      CARDINIA SHIRE   
3  856969545  349      349  MOONEE VALLEY  MOONEE VALLEY CITY   
4  821999870  377      377   YARRA RANGES  YARRA RANGES SHIRE   

           GAZREGN ABSLGACODE  TASK_ID PFI_CR    UFI_OLD      UFI_CR  \
0    S097 1994 p18      21180      NaN    NaN  791830413  2024-09-26   
1  G047 2003 p2972      25710      NaN    NaN  806916803  2023-12-19   
2  G040 2003 p2572      21450      NaN    NaN  812373711  2023-12-19   
3  G050 2007 p3095      25060      NaN    NaN  856190176  2024-09-26   
4  G040 2003 p2574      27450      NaN    NaN  820277769  2024-01-19   

   LABEL_USE_                                           geometry  
0           5  POLYGON ((144.76468 -37.72308, 144.76471 -37.7...  
1           4  POLYGON ((145.11684 -37.68791, 145.11435 -37.6...  
2           2  POLYGON ((145.75495 -38.12652, 145.75490 -38.1...  
3           5  POLYGON ((144.86587 -37.77215, 144.86565 -37.7...  
4           2  POLYGON ((145.79899 -37.82598, 145.79946 -37.8...

In [4]:
lga_data = gdf[['LGA_NAME', 'geometry']]
lga_data['LGA_NAME'] = lga_data['LGA_NAME'].str.lower()
lga_data.rename(columns={'LGA_NAME': 'lga'}, inplace=True)
lga_data

lga  \
0                              brimbank   
1                             nillumbik   
2                              cardinia   
3                         moonee valley   
4                          yarra ranges   
..                                  ...   
82                              benalla   
83                          murrindindi   
84                            mansfield   
85  lake mountain alpine resort (uninc)   
86                              baw baw   

                                             geometry  
0   POLYGON ((144.76468 -37.72308, 144.76471 -37.7...  
1   POLYGON ((145.11684 -37.68791, 145.11435 -37.6...  
2   POLYGON ((145.75495 -38.12652, 145.75490 -38.1...  
3   POLYGON ((144.86587 -37.77215, 144.86565 -37.7...  
4   POLYGON ((145.79899 -37.82598, 145.79946 -37.8...  
..                                                ...  
82  POLYGON ((146.03012 -36.87730, 146.02965 -36.8...  
83  MULTIPOLYGON (((145.84950 -37.52822, 145.84721...  
84  POLYGON ((145.97009 -37.26314, 145.97001 -37.2...  
85  POLYGON ((145.88229 -37.52431, 145.88233 -37.5...  
86  POLYGON ((145.76265 -37.99175, 145.76271 -37.9...  

[87 rows x 2 columns]

In [5]:
lga_data.to_csv('data/curated/lga_coordinates.csv')

In [6]:
lga_data.to_file('data/curated/lga_coord/lga_coordinates.shp')


In [7]:
lga_check = lga_data['lga']

In [8]:
lga_check.to_csv('data/curated/lga_check.csv')

# Population

In [9]:
file_path = 'data/raw/ABS_ERP_LGA2023_1.0.0_.LGA2023..A.csv'
school_quality_df = pd.read_csv(file_path)
school_quality_df.head()

DATAFLOW                    MEASURE: Measure  \
0  ABS:ERP_LGA2023(1.0.0)  ERP: Estimated Resident Population   
1  ABS:ERP_LGA2023(1.0.0)  ERP: Estimated Resident Population   
2  ABS:ERP_LGA2023(1.0.0)  ERP: Estimated Resident Population   
3  ABS:ERP_LGA2023(1.0.0)  ERP: Estimated Resident Population   
4  ABS:ERP_LGA2023(1.0.0)  ERP: Estimated Resident Population   

                 REGION_TYPE: Region Type    REGION: Region FREQ: Frequency  \
0  LGA2023: Local Government Areas (2023)  25710: Nillumbik       A: Annual   
1  LGA2023: Local Government Areas (2023)  25710: Nillumbik       A: Annual   
2  LGA2023: Local Government Areas (2023)  25710: Nillumbik       A: Annual   
3  LGA2023: Local Government Areas (2023)  25710: Nillumbik       A: Annual   
4  LGA2023: Local Government Areas (2023)  25710: Nillumbik       A: Annual   

   TIME_PERIOD: Time Period  OBS_VALUE UNIT_MEASURE: Unit of Measure  \
0                      2013      63174                 PSNS: Persons   
1                      2014      63433                 PSNS: Persons   
2                      2015      63675                 PSNS: Persons   
3                      2016      64174                 PSNS: Persons   
4                      2017      64414                 PSNS: Persons   

   OBS_STATUS: Observation Status  OBS_COMMENT: Observation Comment  \
0                             NaN                               NaN   
1                             NaN                               NaN   
2                             NaN                               NaN   
3                             NaN                               NaN   
4                             NaN                               NaN   

  REPYEAREND: Reporting Year End Day  
0          30-6: Year Ending 30 June  
1          30-6: Year Ending 30 June  
2          30-6: Year Ending 30 June  
3          30-6: Year Ending 30 June  
4          30-6: Year Ending 30 June

In [10]:
selected_columns = school_quality_df[['REGION: Region', 'TIME_PERIOD: Time Period', 'OBS_VALUE']]
selected_columns['REGION: Region'] = selected_columns['REGION: Region'].str.split(': ').str[1]
selected_columns['REGION: Region'] = selected_columns['REGION: Region'].str.lower()

# if lga name ends with "(vic)", remove that
selected_columns['REGION: Region'] = selected_columns['REGION: Region'].str.replace(' \(vic.\)', '', regex=True)

# Rename the columns
selected_columns.rename(columns={
    "REGION: Region": "lga",
    "TIME_PERIOD: Time Period": "year",
    "OBS_VALUE": "lga_population"
}, inplace=True)

selected_columns

lga  year  lga_population
0     nillumbik  2013           63174
1     nillumbik  2014           63433
2     nillumbik  2015           63675
3     nillumbik  2016           64174
4     nillumbik  2017           64414
...         ...   ...             ...
6012     dungog  2019            9329
6013     dungog  2020            9450
6014     dungog  2021            9532
6015     dungog  2022            9721
6016     dungog  2023            9767

[6017 rows x 3 columns]

In [11]:
selected_columns.to_csv('data/curated/lga_pop.csv')

In [12]:
lga_pop = selected_columns

# Merge into VIC LGA list (check for bugs)

In [13]:
# Merge using a left join to ensure all rows from lga_check are preserved
merged_df = pd.merge(lga_check, lga_pop, left_on='lga', right_on='lga', how='left')
merged_df

lga    year  lga_population
0    brimbank  2013.0        196169.0
1    brimbank  2014.0        198468.0
2    brimbank  2015.0        200860.0
3    brimbank  2016.0        204190.0
4    brimbank  2017.0        206116.0
..        ...     ...             ...
872   baw baw  2019.0         54047.0
873   baw baw  2020.0         55753.0
874   baw baw  2021.0         57618.0
875   baw baw  2022.0         59277.0
876   baw baw  2023.0         60644.0

[877 rows x 3 columns]

In [14]:
merged_df = merged_df.sort_values(by=['lga', 'year'])
merged_df

lga    year  lga_population
463        alpine  2013.0         12283.0
464        alpine  2014.0         12355.0
465        alpine  2015.0         12429.0
466        alpine  2016.0         12578.0
467        alpine  2017.0         12687.0
..            ...     ...             ...
536  yarriambiack  2019.0          6675.0
537  yarriambiack  2020.0          6641.0
538  yarriambiack  2021.0          6510.0
539  yarriambiack  2022.0          6437.0
540  yarriambiack  2023.0          6389.0

[877 rows x 3 columns]

In [15]:
#merged_df.to_csv('data/curated/lga_pop_bugcheck.csv')

In [16]:
merged_df = merged_df.dropna() # remove rows with empty population (national parks etc.)
merged_df = merged_df.reset_index(drop=True) # remove index

In [17]:
merged_df

lga    year  lga_population
0          alpine  2013.0         12283.0
1          alpine  2014.0         12355.0
2          alpine  2015.0         12429.0
3          alpine  2016.0         12578.0
4          alpine  2017.0         12687.0
..            ...     ...             ...
864  yarriambiack  2019.0          6675.0
865  yarriambiack  2020.0          6641.0
866  yarriambiack  2021.0          6510.0
867  yarriambiack  2022.0          6437.0
868  yarriambiack  2023.0          6389.0

[869 rows x 3 columns]

In [18]:
merged_df.to_csv('data/curated/lga_pop.csv', index=False)

## 2024-2027 Population Forecast

In [19]:
data = merged_df

# Get a list of unique LGAs
lgas = data['lga'].unique()

# Check the range of years to ensure data consistency for forecasting
year_range = data['year'].min(), data['year'].max()

lgas, year_range

(array(['alpine', 'ararat', 'ballarat', 'banyule', 'bass coast', 'baw baw',
        'bayside', 'benalla', 'boroondara', 'brimbank', 'buloke',
        'campaspe', 'cardinia', 'casey', 'central goldfields',
        'colac otway', 'corangamite', 'darebin', 'east gippsland',
        'frankston', 'gannawarra', 'glen eira', 'glenelg', 'golden plains',
        'greater bendigo', 'greater dandenong', 'greater geelong',
        'greater shepparton', 'hepburn', 'hindmarsh', 'hobsons bay',
        'horsham', 'hume', 'indigo', 'kingston', 'knox', 'latrobe',
        'loddon', 'macedon ranges', 'manningham', 'mansfield',
        'maribyrnong', 'maroondah', 'melbourne', 'melton', 'merri-bek',
        'mildura', 'mitchell', 'moira', 'monash', 'moonee valley',
        'moorabool', 'mornington peninsula', 'mount alexander', 'moyne',
        'murrindindi', 'nillumbik', 'northern grampians', 'port phillip',
        'pyrenees', 'queenscliffe', 'south gippsland',
        'southern grampians', 'stonnington',

In [20]:
from statsmodels.tsa.arima.model import ARIMA
import numpy as np

# Create a dictionary to store the forecasts
forecasts = {}

# Adjust the forecast retrieval process and rerun the forecast for 2024
for lga in lgas:
    # Filter the data for the current LGA
    lga_data = data[data['lga'] == lga].set_index('year')
    
    # Ensure no missing values and only population is used
    lga_population = lga_data['lga_population'].dropna()
    
    # Fit ARIMA model (Auto ARIMA can be done to select the best order, but here we use a simple approach)
    model = ARIMA(lga_population, order=(1, 1, 1))  # Using a (1,1,1) ARIMA model for simplicity
    model_fit = model.fit()
    
    # Forecast the population for 2024 (1 year ahead)
    forecast = model_fit.forecast(steps=1)
    
    # Store the forecast
    forecasts[lga] = forecast.iloc[0]  # Use iloc to correctly retrieve the forecast

# Convert forecasts to a DataFrame for easy viewing
forecast_df = pd.DataFrame(list(forecasts.items()), columns=['lga', 'forecast_2024'])
forecast_df

/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provide

lga  forecast_2024
0         alpine   13182.919946
1         ararat   11633.718762
2       ballarat  120111.150937
3        banyule  129732.250887
4     bass coast   43863.453426
..           ...            ...
74       wodonga   44783.676963
75       wyndham  337384.401708
76         yarra   97727.776726
77  yarra ranges  158928.458022
78  yarriambiack    6351.316010

[79 rows x 2 columns]

In [21]:
# Add the forecast results for 2024 into the initial dataset
forecast_2024_df = pd.DataFrame({'lga': forecast_df['lga'], 'year': 2024, 'lga_population': forecast_df['forecast_2024']})

# Append the forecast results to the original dataset
pop_df = pd.concat([data, forecast_2024_df], ignore_index=True)
pop_df = pop_df.sort_values(by=['lga', 'year'])

In [22]:
pop_df.head(20)

lga    year  lga_population
0    alpine  2013.0    12283.000000
1    alpine  2014.0    12355.000000
2    alpine  2015.0    12429.000000
3    alpine  2016.0    12578.000000
4    alpine  2017.0    12687.000000
5    alpine  2018.0    12822.000000
6    alpine  2019.0    12904.000000
7    alpine  2020.0    13054.000000
8    alpine  2021.0    13166.000000
9    alpine  2022.0    13191.000000
10   alpine  2023.0    13182.000000
869  alpine  2024.0    13182.919946
11   ararat  2013.0    11490.000000
12   ararat  2014.0    11560.000000
13   ararat  2015.0    11644.000000
14   ararat  2016.0    11745.000000
15   ararat  2017.0    11796.000000
16   ararat  2018.0    11805.000000
17   ararat  2019.0    11838.000000
18   ararat  2020.0    11918.000000

In [23]:
pop_df.to_csv('data/curated/lga_pop_2024.csv', index=False)

In [24]:
from statsmodels.tsa.arima.model import ARIMA
import numpy as np
import pandas as pd

# Create a dictionary to store the forecasts
forecasts = {}

# Adjust the forecast retrieval process and rerun the forecast for 2024 to 2027
for lga in lgas:
    # Filter the data for the current LGA
    lga_data = data[data['lga'] == lga].set_index('year')
    
    # Ensure no missing values and only population is used
    lga_population = lga_data['lga_population'].dropna()
    
    # Fit ARIMA model (Auto ARIMA can be done to select the best order, but here we use a simple approach)
    model = ARIMA(lga_population, order=(1, 1, 1))  # Using a (1,1,1) ARIMA model for simplicity
    model_fit = model.fit()
    
    # Forecast the population for 2024 to 2027 (4 years ahead)
    forecast = model_fit.forecast(steps=4)
    
    # Store the forecasts for 2024, 2025, 2026, and 2027
    forecasts[lga] = forecast

# Convert forecasts to a DataFrame for easy viewing
forecast_df = pd.DataFrame(forecasts).T
forecast_df.columns = ['forecast_2024', 'forecast_2025', 'forecast_2026', 'forecast_2027']

# Reset index and rename the first column to 'lga'
forecast_df = forecast_df.reset_index().rename(columns={'index': 'lga'})

forecast_df

/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provide

lga  forecast_2024  forecast_2025  forecast_2026  forecast_2027
0         alpine   13182.919946   13183.714226   13184.400005   13184.992106
1         ararat   11633.718762   11597.322067   11570.441260   11550.588419
2       ballarat  120111.150937  122085.200837  124059.149705  126032.997546
3        banyule  129732.250887  129836.179714  129919.105837  129985.273637
4     bass coast   43863.453426   44997.861040   46132.222844   47266.538840
..           ...            ...            ...            ...            ...
74       wodonga   44783.676963   45266.519980   45725.743849   46162.503943
75       wyndham  337384.401708  350680.580394  363975.536171  377269.269151
76         yarra   97727.776726   97932.238071   98081.658798   98190.855740
77  yarra ranges  158928.458022  159121.961561  159281.664434  159413.470849
78  yarriambiack    6351.316010    6321.437041    6297.746535    6278.962751

[79 rows x 5 columns]

In [25]:
# Add the forecast results
forecast_2024_df = pd.DataFrame({'lga': forecast_df['lga'], 'year': 2024, 'lga_population': forecast_df['forecast_2024']})
forecast_2025_df = pd.DataFrame({'lga': forecast_df['lga'], 'year': 2025, 'lga_population': forecast_df['forecast_2025']})
forecast_2026_df = pd.DataFrame({'lga': forecast_df['lga'], 'year': 2026, 'lga_population': forecast_df['forecast_2026']})
forecast_2027_df = pd.DataFrame({'lga': forecast_df['lga'], 'year': 2027, 'lga_population': forecast_df['forecast_2027']})

In [26]:

# Append the forecast results to the original dataset
pop_df = pd.concat([data, forecast_2024_df, forecast_2025_df, forecast_2026_df, forecast_2027_df], ignore_index=True)
pop_df = pop_df.sort_values(by=['lga', 'year'])
pop_df

lga    year  lga_population
0           alpine  2013.0    12283.000000
1           alpine  2014.0    12355.000000
2           alpine  2015.0    12429.000000
3           alpine  2016.0    12578.000000
4           alpine  2017.0    12687.000000
...            ...     ...             ...
868   yarriambiack  2023.0     6389.000000
947   yarriambiack  2024.0     6351.316010
1026  yarriambiack  2025.0     6321.437041
1105  yarriambiack  2026.0     6297.746535
1184  yarriambiack  2027.0     6278.962751

[1185 rows x 3 columns]

In [27]:
pop_df.to_csv('data/curated/lga_pop.csv', index=False)

## Forecast accuracy

In [28]:
from sklearn.metrics import mean_absolute_error, mean_squared_error
import numpy as np

# Initialize lists to store error metrics for each LGA
mae_list = []
mse_list = []
rmse_list = []

# Iterate through each LGA and compute error metrics
for lga in lgas:
    # Filter the data for the current LGA
    lga_data = data[data['lga'] == lga].set_index('year')
    
    # Ensure no missing values and only population is used
    lga_population = lga_data['lga_population'].dropna()
    
    # Fit ARIMA model
    model = ARIMA(lga_population, order=(1, 1, 1))
    model_fit = model.fit()
    
    # Perform in-sample forecast (fitted values)
    fitted_values = model_fit.fittedvalues
    
    # Compute errors
    mae = mean_absolute_error(lga_population[1:], fitted_values)
    mse = mean_squared_error(lga_population[1:], fitted_values)
    rmse = np.sqrt(mse)
    
    # Store the errors
    mae_list.append(mae)
    mse_list.append(mse)
    rmse_list.append(rmse)

# Create a DataFrame to display the accuracy metrics
accuracy_df = pd.DataFrame({
    'LGA': lgas,
    'MAE': mae_list,
    'MSE': mse_list,
    'RMSE': rmse_list
})
accuracy_df

/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. 

LGA           MAE           MSE          RMSE
0         alpine   1150.534531  1.371773e+07   3703.745522
1         ararat   1084.482824  1.200461e+07   3464.766372
2       ballarat   9161.932310  8.793061e+08  29653.096486
3        banyule  12086.477959  1.406377e+09  37501.695313
4     bass coast   3149.022490  8.936943e+07   9453.540633
..           ...           ...           ...           ...
74       wodonga   3464.777536  1.283571e+08  11329.477935
75       wyndham  18776.192052  3.292993e+09  57384.602641
76         yarra   8890.130335  6.461417e+08  25419.318229
77  yarra ranges  14271.812316  2.070686e+09  45504.796410
78  yarriambiack    673.571243  4.498327e+06   2120.925874

[79 rows x 4 columns]

In [29]:
# Calculate overall accuracy metrics (mean of MAE, MSE, RMSE across all LGAs)
overall_mae = np.mean(mae_list)
overall_mse = np.mean(mse_list)
overall_rmse = np.mean(rmse_list)

overall_accuracy = {
    'Overall MAE': overall_mae,
    'Overall MSE': overall_mse,
    'Overall RMSE': overall_rmse
}

overall_accuracy

{'Overall MAE': 7196.6728439181425,
 'Overall MSE': 880403212.734419,
 'Overall RMSE': 22059.421859065733}

The overall MAE of 7,196.67 means that on average the model’s population predictions were off by around 7,197 people per LGA.

The overall RMSE of 22,059.42 indicates that the typical prediction error across all LGAs is around 22,059 people.

While some LGAs had relatively small errors, others had much larger deviations, which inflated the overall metrics.

# Crime

In [30]:
# Load the Excel file to inspect the available sheets
file_path = 'data/raw/Data_Tables_LGA_Recorded_Offences_Year_Ending_June_2024.xlsx'
xls = pd.ExcelFile(file_path)

# List all sheet names to identify "Table 02"
sheet_names = xls.sheet_names
sheet_names

['Contents',
 'Footnotes',
 'Table 01',
 'Table 02',
 'Table 03',
 'Table 04',
 'Table 05',
 'Table 06']

In [31]:
df_raw = pd.read_excel(file_path, sheet_name='Table 01')
df_raw

Year Year ending                                    Police Region  \
0    2024        June                               1 North West Metro   
1    2024        June                               1 North West Metro   
2    2024        June                               1 North West Metro   
3    2024        June                               1 North West Metro   
4    2024        June                               1 North West Metro   
..    ...         ...                                              ...   
865  2015        June                                        4 Western   
866  2015        June  Justice Institutions and Immigration Facilities   
867  2015        June  Justice Institutions and Immigration Facilities   
868  2015        June                               Unincorporated Vic   
869  2015        June                               Unincorporated Vic   

                                Local Government Area  Offence Count  \
0                                             Banyule           8929   
1                                            Brimbank          17551   
2                                             Darebin          14621   
3                                         Hobsons Bay           6416   
4                                                Hume          20132   
..                                                ...            ...   
865                                             Total          74737   
866   Justice Institutions and Immigration Facilities           1041   
867                                             Total           1041   
868                                Unincorporated Vic            127   
869                                             Total            127   

     Rate per 100,000 population  
0                    6708.126083  
1                    8856.435232  
2                    9172.670623  
3                    6760.139376  
4                    7444.876870  
..                           ...  
865                  7857.458261  
866                          NaN  
867                          NaN  
868                          NaN  
869                          NaN  

[870 rows x 6 columns]

In [32]:
nan_rate_rows = df_raw[df_raw['Rate per 100,000 population'].isna()]
nan_rate_rows.head(10)

Year Year ending                                    Police Region  \
83   2024        June  Justice Institutions and Immigration Facilities   
84   2024        June  Justice Institutions and Immigration Facilities   
85   2024        June                               Unincorporated Vic   
86   2024        June                               Unincorporated Vic   
170  2023        June  Justice Institutions and Immigration Facilities   
171  2023        June  Justice Institutions and Immigration Facilities   
172  2023        June                               Unincorporated Vic   
173  2023        June                               Unincorporated Vic   
257  2022        June  Justice Institutions and Immigration Facilities   
258  2022        June  Justice Institutions and Immigration Facilities   

                                Local Government Area  Offence Count  \
83    Justice Institutions and Immigration Facilities           1798   
84                                              Total           1798   
85                                 Unincorporated Vic            142   
86                                              Total            142   
170   Justice Institutions and Immigration Facilities           2440   
171                                             Total           2440   
172                                Unincorporated Vic            101   
173                                             Total            101   
257   Justice Institutions and Immigration Facilities           1799   
258                                             Total           1799   

     Rate per 100,000 population  
83                           NaN  
84                           NaN  
85                           NaN  
86                           NaN  
170                          NaN  
171                          NaN  
172                          NaN  
173                          NaN  
257                          NaN  
258                          NaN

In [33]:
df = df_raw[['Local Government Area', 'Year', 'Rate per 100,000 population']]
df['Local Government Area'] = df['Local Government Area'].str.strip()  # Remove any leading/trailing whitespace
df = df[~df['Local Government Area'].isin(['Total', 'Justice Institutions and Immigration Facilities', 'Unincorporated Vic'])]
df.head()

Local Government Area  Year  Rate per 100,000 population
0               Banyule  2024                  6708.126083
1              Brimbank  2024                  8856.435232
2               Darebin  2024                  9172.670623
3           Hobsons Bay  2024                  6760.139376
4                  Hume  2024                  7444.876870

In [34]:
df['Local Government Area'] = df['Local Government Area'].str.lower()  # Convert to lowercase
df = df.sort_values(by=['Local Government Area', 'Year'])
df

Local Government Area  Year  Rate per 100,000 population
798                alpine  2015                  3572.290611
711                alpine  2016                  3378.915567
624                alpine  2017                  2987.309845
537                alpine  2018                  3899.547652
450                alpine  2019                  4657.470552
..                    ...   ...                          ...
429          yarriambiack  2020                  8312.001205
342          yarriambiack  2021                  6298.003072
255          yarriambiack  2022                  6571.384185
168          yarriambiack  2023                  7168.570981
81           yarriambiack  2024                  8609.398034

[790 rows x 3 columns]

In [35]:
df.to_csv('data/curated/crime_by_lga_2024.csv')

### Forecast to 2027

In [36]:
crime_data = df

In [37]:
from statsmodels.tsa.arima.model import ARIMA

# Prepare data for ARIMA forecasting
lgas = crime_data['Local Government Area'].unique()

# Create a dictionary to store the forecasts
forecasts = {}

# Adjust the forecast retrieval process and rerun the forecast for 2025 to 2027
for lga in lgas:
    # Filter the data for the current LGA
    lga_data = crime_data[crime_data['Local Government Area'] == lga].set_index('Year')
    
    # Ensure no missing values and only 'Rate per 100,000 population' is used
    lga_population = lga_data['Rate per 100,000 population'].dropna()
    
    # Fit ARIMA model (Auto ARIMA could be used, but we'll use a simple (1,1,1) model)
    model = ARIMA(lga_population, order=(1, 1, 1))
    model_fit = model.fit()
    
    # Forecast the population for 2025 to 2027 (3 years ahead)
    forecast = model_fit.forecast(steps=3)
    
    # Store the forecasts for 2025, 2026, and 2027
    forecasts[lga] = forecast

# Convert forecasts to a DataFrame for easy viewing
forecast_df = pd.DataFrame(forecasts).T
forecast_df.columns = ['forecast_2025', 'forecast_2026', 'forecast_2027']

# Reset index and rename the first column to 'lga'
forecast_df = forecast_df.reset_index().rename(columns={'index': 'lga'})

/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provided and will be ignored when e.g. forecasting.
  self._init_dates(dates, freq)
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:836: ValueWarning: No supported index is available. Prediction results will be given with an integer index beginning at `start`.
  return get_prediction_index(
/home/kevin/.local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py:473: ValueWarning: An unsupported index was provide

In [38]:
forecast_df

lga  forecast_2025  forecast_2026  forecast_2027
0         alpine    3763.089427    3723.349452    3695.776916
1         ararat   12743.128051   11863.438666   12743.124028
2       ballarat   10734.777347   10751.582053   10754.143243
3        banyule    6667.336609    6634.830579    6608.925809
4     bass coast    7311.166938    7112.388535    6999.153570
..           ...            ...            ...            ...
74       wodonga   10694.339194   10530.061799   10668.647648
75       wyndham    5871.766141    5819.258631    5795.071476
76         yarra   13772.111507   13617.871127   13721.981767
77  yarra ranges    4306.714000    4317.134961    4316.254802
78  yarriambiack    8391.308656    8525.497956    8442.931953

[79 rows x 4 columns]

In [39]:
# Convert forecast data to a format similar to the original dataset
forecast_2025_df = pd.DataFrame({'Local Government Area': forecast_df['lga'], 'Year': 2025, 'Rate per 100,000 population': forecast_df['forecast_2025']})
forecast_2026_df = pd.DataFrame({'Local Government Area': forecast_df['lga'], 'Year': 2026, 'Rate per 100,000 population': forecast_df['forecast_2026']})
forecast_2027_df = pd.DataFrame({'Local Government Area': forecast_df['lga'], 'Year': 2027, 'Rate per 100,000 population': forecast_df['forecast_2027']})

# Append the forecasted data to the original dataset
combined_df = pd.concat([crime_data, forecast_2025_df, forecast_2026_df, forecast_2027_df], ignore_index=True)

# Sort the combined data by 'lga' and 'year'
combined_df = combined_df.sort_values(by=['Local Government Area', 'Year'])

# Display the combined dataframe
combined_df


Local Government Area  Year  Rate per 100,000 population
0                   alpine  2015                  3572.290611
1                   alpine  2016                  3378.915567
2                   alpine  2017                  2987.309845
3                   alpine  2018                  3899.547652
4                   alpine  2019                  4657.470552
...                    ...   ...                          ...
788           yarriambiack  2023                  7168.570981
789           yarriambiack  2024                  8609.398034
868           yarriambiack  2025                  8391.308656
947           yarriambiack  2026                  8525.497956
1026          yarriambiack  2027                  8442.931953

[1027 rows x 3 columns]

In [40]:
combined_df.to_csv('data/curated/crime_by_lga.csv')